lib/utils.pm

```diff
@@ -3136,4 +3139,43 @@ sub enable_console_kernel_log {
     record_info("Content of /proc/sys/kernel/printk", script_output("cat /proc/sys/kernel/printk"));
 }
 
+=head2 is_ram0_root_device
+
+ is_ram0_root_device;
+
+When using initrd, the system typically boots the boot loader and loads the kernel
+and the initial RAM disk. The kernel converts initrd into a RAM disk and frees the
+memory used by initrd. If the root device is not /dev/ram0, the old and deprecated
+change_root procedure is followed instead of pivot_root procedure.
+=cut
+
+sub is_ram0_root_device {
+    return 0 if (check_var('CHANGE_ROOT', 1) or !get_var('PIVOT_ROOT'));
+    return 1;
+}
+
+=head2 is_disk_image
+
+ is_disk_image;
+
+Identify whether test runs with linux disk image built by kiwi or similar programs
+=cut
+
+sub is_disk_image {
+    return 1 if (get_var('HDD_1'));
```

Comment:
Many tests have HDD_1 defined, but that does not necessarily mean the variable represents a bootable image

Reply:
It does not have to bootable I think, as long as they are used for installation and they are pre-built ones.

Reply:
It can be used even for an empty drive. The subroutine in this form is ambiguous IMHO

Reply:
> It can be used even for an empty drive. The subroutine in this form is ambiguous IMHO

Could you help share the job link ?

Reply:
I added ```is_sle_micro``` to make it more strict. It can also be extended in the future.